runtime/sema/checker.go

```diff
@@ -204,12 +204,12 @@ func (checker *Checker) CurrentPurityScope() PurityCheckScope {
 	return checker.purityCheckScopes[len(checker.purityCheckScopes)-1]
 }
 
-func (checker *Checker) PushNewPurityScope(enforce bool) {
+func (checker *Checker) PushNewPurityScope(enforce bool, depth int) {
 	checker.purityCheckScopes = append(
 		checker.purityCheckScopes,
 		PurityCheckScope{
```

Comment:
Didn't think of this in the previous PR: Can the purity scopes be combined with the function activations? Or are they not always mapped 1:1? That could reduce the complexity a bit

Reply:
No; they are specifically separate so we can push a pure scope when we enter a precondition or postcondition in the next PR.